tests/components/sql/__init__.py

```diff
@@ -27,6 +27,8 @@
     CONF_QUERY: "SELECT 5 as value",
     CONF_COLUMN_NAME: "value",
     CONF_UNIT_OF_MEASUREMENT: "MiB",
+    CONF_DEVICE_CLASS: "data_size",
```

Comment:
Should add a test to cover the new scenarios
- Setup with `"None"` as `device_class` and `state_class`
- Change to a `device_class` and `state_class`
- Change back to `"None"`

Reply:
Added at https://github.com/home-assistant/core/pull/95020/commits/ef1355796ad79acb2e649570b096ef0bcbb4b54b